Outfitter Updates and Code Refactor (#10743)
endless-sky/endless-sky: data/_ui/help.txt

help "shipyard"
	`Here, you can buy new ships.`
	`Use your scroll wheel or click and drag, to scroll the view.`
	`As in the trading and outfitter panels, you can hold down Shift, Control, or Alt to buy 5, 20, or 500 ships at once, or multiple keys for larger amounts.`
	`You can use the shortcuts B to Buy, S to Sell the ship and outfits and R to sell the ship and retain the outfits.`
	`If the planet has an outfitter, the retained outfits will go to planetary storage, otherwise they will go into cargo.`
	`You can also press F to search the shipyard for a specific ship.`

help "cargo management"
	`Sometimes you may want to buy outfits directly into your cargo hold instead of having them installed. Perhaps a mission has asked you to retrieve some outfits or perhaps you want to carry extra ammunition to refill your ships somewhere that does not stock those weapons.`
	`You can use the following keyboard shortcuts or the buttons to move outfits around:`
	` - "B" to buy and install outfits`
	` - "S" to sell outfits from cargo, planetary storage, or your ship (in that order)`
	` - "U" to uninstall outfits from your ship to your cargo hold, with spillover ending up in planetary storage`

> 	` - "I" to install outfits from cargo and/or storage (in that order)`

Reviewer: ```suggestion
	` - "I" to install outfits from cargo or storage (in that order)`
```

Author: It's "and/or" intentionally:

E.g. Javelin limit is 300 right now:
![2024-11-04 07_20_16-Screenshot from 2024-11-04 07-19-48 png](https://github.com/user-attachments/assets/74115731-1a47-4ce6-a9f1-7112ece7650d)   Press Alt-I (install 500)   ![2024-11-04 07_20_45-Screenshot from 2024-11-04 07-20-01 png](https://github.com/user-attachments/assets/d81e1e92-680d-4fff-81fe-bc146374bd58)

It will perform the "Install" command and use all available sources, in priority order as described.  15 from cargo, then 35 from storage.

Author: @TheGiraffe3, I'd like to hear your feedback before marking this as resolved.

	`Here, you can buy new equipment for your ship. Your ship has a limited amount of "outfit space," and most outfits use up some of that space.`
	`Some types of outfits have other requirements as well. For example, only some of your outfit space can be used for engines or weapons; this is your ship's "engine capacity" and "weapon capacity." Guns and missile launchers also require a free "gun port," and turrets require a free "turret mount." Also, missiles can only be bought if you have the right launcher installed.`
	`Use your scroll wheel or click and drag, to scroll the view.`

> 	`As in the trading and shipyard panels, you can hold down Shift, Control, or Alt to buy, sell or move 5, 20, or 500 of an outfit at once, or multiple keys for larger amounts.`

Reviewer 2: ```suggestion
	`As in the trading and shipyard panels, you can hold down Shift, Control, or Alt to buy, sell, or move 5, 20, or 500 of an outfit at once, or multiple keys for larger amounts.`
```

	`You can also press F to search the outfitter for a specific item.`

help "shipyard"
	`Here, you can buy new ships.`
	`Use your scroll wheel or click and drag, to scroll the view.`
	`As in the trading and outfitter panels, you can hold down Shift, Control, or Alt to buy 5, 20, or 500 ships at once, or multiple keys for larger amounts.`

> 	`You can use the shortcuts B to Buy, S to Sell the ship and outfits and R to sell the ship and retain the outfits if the planet has an outfitter.`

Reviewer 2: ```suggestion
	`You can use the shortcuts B to Buy, S to Sell the ship and outfits, and R to sell the ship and retain the outfits if the planet has an outfitter.`
```

Reviewer 2: Do we need this tooltip? The `cargo managment` tooltip right under it gives the same information.

	`You can also press F to search the shipyard for a specific ship.`

help "cargo management"

> 	`Sometimes you may want to buy outfits directly into your cargo hold instead of having them installed. Perhaps a mission has asked you to retrieve some outfits or perhaps you want to carry extra ammunition to refill your ships somewhere that does not stock those weapons.`

Reviewer 2: ```suggestion
	`Sometimes you may want to buy outfits directly into your cargo hold instead of having them installed. Perhaps a mission has asked you to retrieve some outfits, or perhaps you want to carry extra ammunition to refill your ships somewhere that does not stock those weapons.`
```

	`Outfits that are in your cargo hold can be installed at any outfitter without cost.`

help "uninstalling and storage"
	`Using planetary storage can be useful for if you want to shuffle outfits between ships, store outfits for future use, or simply try different outfits.`
	`Outfits that are in planetary storage or cargo can be installed without cost.`
	`If you sell an outfit at an outfitter, then you can immediately buy it back for the same price that you sold it, so long as you do not leave the planet.`
	`This means that you do not necessarily need to worry about uninstalling outfits into planetary storage when you are changing outfits around.`

help "outfitter with multiple ships"
	`You can manage outfits on multiple ships at the same time. Ctrl+click a ship icon to add it to your selection. You can also use Shift+click to select a range of ships.`
	`When you have more than one ship selected, you can install or remove outfits from all of those ships at once. Additionally, all selected ships with the currently selected outfit installed are marked with a pointer next to their sidebar icon.`

> 	`Escorts can be saved into control groups by pressing Control+number key. All selected escorts will then be bound to that number key, selecting the group when pressed.`

Reviewer 2: ```suggestion
	`Escorts can be saved into control groups by pressing a number key while holding Control. All selected escorts will then be bound to that number key, and can be selected again at any time by pressing it.`
```
